tests/manual-test-cases/Group21-Registries/21-1-Whitelist.robot

```diff
@@ -0,0 +1,318 @@
+    Should Be Equal As Integers  ${rc}  0
+    Log  ${output}
+#    ${rc}  ${output}=  Run And Return Rc And Output  docker tag busybox ${HTTP_HARBOR_IP}/library/busybox
+#    Should Be Equal As Integers  ${rc}  0
+#    Log  ${output}
+#    ${rc}  ${output}=  Run And Return Rc And Output  docker push ${HTTP_HARBOR_IP}/library/busybox   
+#    Should Be Equal As Integers  ${rc}  0
+#    Log  ${output}
+
+
+*** Test Cases ***
+Basic Whitelisting
+    Run Keyword If  '%{HOST_TYPE}' == 'ESXi'  Pass Execution  Test skipped on ESXi as Harbor is only supported on VC
+    
+    # Attempt to cleanup old/canceled tests
+    Run Keyword And Ignore Error  Cleanup Dangling VMs On Test Server
+    Run Keyword And Ignore Error  Cleanup Datastore On Test Server
+
+    # Install VCH with registry CA for whitelisted registry
+    ${output}=  Install VIC Appliance To Test Server  vol=default --whitelist-registry=%{HTTPS_HARBOR_IP} --registry-ca=./ca.crt
+    Should Contain  ${output}  Secure registry %{HTTPS_HARBOR_IP} confirmed
+    Should Contain  ${output}  Whitelist registries =
+
+    # Check docker info for whitelist info
+    ${rc}  ${output}=  Run And Return Rc And Output  docker %{VCH-PARAMS} info
+    Should Be Equal As Integers  ${rc}  0
+    Should Contain  ${output}  Registry Whitelist Mode: enabled
+    Should Contain  ${output}  Whitelisted Registries:
+    Should Contain  ${output}  Registry: registry-1.docker.io
+
+    # Try to login and pull from the HTTPS whitelisted registry (should succeed)
+    ${rc}  ${output}=  Run And Return Rc And Output  docker %{VCH-PARAMS} login -u admin -p %{TEST_PASSWORD} %{HTTPS_HARBOR_IP}
+    Should Contain  ${output}  Succeeded
+    Should Be Equal As Integers  ${rc}  0
+    ${rc}  ${output}=  Run And Return Rc And Output  docker %{VCH-PARAMS} pull %{HTTPS_HARBOR_IP}/library/photon:1.0
+    Should Be Equal As Integers  ${rc}  0
+
+    # Try to login and pull from the HTTPS whitelisted registry with :443 tacked on at the end (should succeed)
+    ${rc}  ${output}=  Run And Return Rc And Output  docker %{VCH-PARAMS} login -u admin -p %{TEST_PASSWORD} %{HTTPS_HARBOR_IP}:443
+    Should Contain  ${output}  Succeeded
+    Should Be Equal As Integers  ${rc}  0
+    ${rc}  ${output}=  Run And Return Rc And Output  docker %{VCH-PARAMS} pull %{HTTPS_HARBOR_IP}:443/library/photon:1.0
+    Should Be Equal As Integers  ${rc}  0
+
+    # Try to login and pull from docker hub (should fail)
+    ${rc}  ${output}=  Run And Return Rc And Output  docker %{VCH-PARAMS} login --username=victest --password=vmware!123
+    Should Be Equal As Integers  ${rc}  1
+    Should Contain  ${output}  Access denied to unauthorized registry
+    ${rc}  ${output}=  Run And Return Rc And Output  docker %{VCH-PARAMS} pull victest/busybox
+    Should Be Equal As Integers  ${rc}  1
+    Should Contain  ${output}  Access denied to unauthorized registry
+
+    Cleanup VIC Appliance On Test Server
```

Comment:
Any failure within the test case will leave this hanging. Better to include this in your overall cleanup and/or test cleanup keyword.  You can have both a suite teardown and a test teardown, so you could leave the harbor instances through the tests but cleanup each vic appliance.